src/python/pants/option/arg_splitter.py

```diff
@@ -206,12 +206,14 @@ def assign_flag_to_scope(flg: str, default_scope: str) -> None:
 
   @staticmethod
   def spec(arg: str) -> bool:
-    """Does arg 'look like' a Spec (rather than a goal name).
+    """Does arg 'look like' a spec (rather than a goal name).
 
     An arg is a spec if it looks like an AddressSpec or a FilesystemSpec.
     In the future we can expand this heuristic to support other kinds of specs, such as URLs.
     """
-    return os.path.sep in arg or ':' in arg or '*' in arg or Path(arg).exists()
+    return (
+      os.path.sep in arg or ':' in arg or arg.startswith('!') or '*' in arg or Path(arg).exists()
```

Comment:
Is it possible to split this vertically, e.g. 
```suggestion
      os.path.sep in arg 
      or ':' in arg 
      or arg.startswith('!') 
      or '*' in arg 
      or Path(arg).exists()
```

Reply:
We could also replace this with:
```python
return re.search('|'.join([re.escape(os.path.sep), ':', '^!', r'\*']), arg) or Path(arg).exists()
```

Reply:
Thanks for the regex snippet!!

Reply:
Actually, not gonna go with regex because of the performance hit compared to using `x in xs`. Will still deduplicate this a bit, tho.